apps/web/utils/actions/invitation.ts

```diff
@@ -0,0 +1,95 @@
+"use server";
+
+import { actionClientUser } from "@/utils/actions/safe-action";
+import { handleInvitationBody } from "@/utils/actions/invitation.validation";
+import { betterAuthConfig } from "@/utils/auth";
+import { SafeError } from "@/utils/error";
+import { headers } from "next/headers";
+import prisma from "@/utils/prisma";
+import type { Invitation } from "better-auth/plugins";
+
+export const handleInvitationAction = actionClientUser
+  .metadata({ name: "handleInvitation" })
+  .schema(handleInvitationBody)
+  .action(async ({ parsedInput: { invitationId } }) => {
+    const session = await betterAuthConfig.api.getSession({
+      headers: await headers(),
+    });
+
+    if (!session) {
+      throw new SafeError("User not authenticated", 401);
+    }
+
+    let invitation: Invitation;
+    try {
+      invitation = await betterAuthConfig.api.getInvitation({
+        query: {
+          id: invitationId,
+        },
+        headers: await headers(),
```

Comment:
id potentially do await headers() just one time above, as we call it again below.